hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/crypto/validators/CryptoCreateChecks.java

```diff
@@ -195,84 +166,62 @@ public static boolean onlyAliasProvided(CryptoCreateTransactionBody op) {
-                return INVALID_ALIAS_KEY;
-            }
-
-            final var keyValidity = validateKey(op);
-            if (keyValidity != OK) {
-                return keyValidity;
-            }
-
-            var isEvmAddressUsedCheck = isUsedAsAliasCheck(op.getAlias());
-            if (isEvmAddressUsedCheck != OK) {
-                return isEvmAddressUsedCheck;
-            }
-
-            return validateEcdsaKey(
-                    op.getKey().getECDSASecp256K1(), op.getAlias().toByteArray());
-        }
-
         if (!dynamicProperties.isCryptoCreateWithAliasEnabled()) {
             return NOT_SUPPORTED;
         }
         final var keyValidity = validateKey(op);
         if (keyValidity != OK) {
             return keyValidity;
         }
+        final var alias = op.getAlias();
+        if (alias.size() == EVM_ADDRESS_SIZE) {
+            return validateEvmAddressAlias(alias);
+        } else {
+            return validatePublicKeyAlias(alias);
+        }
+    }
 
-        if (!isSerializedProtoKey(op.getAlias())) {
+    private ResponseCodeEnum validateOnlyAliasProvidedCase(final CryptoCreateTransactionBody op) {
+        if (!dynamicProperties.isCryptoCreateWithAliasEnabled()) {
+            return NOT_SUPPORTED;
+        }
+        final var alias = op.getAlias();
+        if (alias.size() == EVM_ADDRESS_SIZE) {
             return INVALID_ALIAS_KEY;
+        } else {
+            return validatePublicKeyAlias(alias);
         }
+    }
 
-        var keyFromAlias = asPrimitiveKeyUnchecked(op.getAlias());
-        var key = op.getKey();
-        if ((!key.getEd25519().isEmpty() || !key.getECDSASecp256K1().isEmpty()) && !key.equals(keyFromAlias)) {
+    private ResponseCodeEnum validatePublicKeyAlias(final ByteString alias) {
+        if (!isSerializedProtoKey(alias)) {
             return INVALID_ALIAS_KEY;
         }
-
-        var isAliasUsedCheck = isUsedAsAliasCheck(op.getAlias());
-
+        final var isAliasUsedCheck = isUsedAsAliasCheck(alias);
         if (isAliasUsedCheck != OK) {
             return isAliasUsedCheck;
         }
-
+        final var keyFromAlias = asPrimitiveKeyUnchecked(alias);
         if (!keyFromAlias.getECDSASecp256K1().isEmpty()) {
-
             return tryToRecoverEVMAddressAndCheckValidity(
                     keyFromAlias.getECDSASecp256K1().toByteArray());
         }
-
         return OK;
     }
 
-    private ResponseCodeEnum validateOnlyAliasProvidedCase(final CryptoCreateTransactionBody op) {
-        if (op.getAlias().size() == EVM_ADDRESS_SIZE) {
-            return INVALID_ALIAS_KEY;
-        }
-        if (!dynamicProperties.isCryptoCreateWithAliasEnabled()) {
-            return NOT_SUPPORTED;
-        }
-        if (!isSerializedProtoKey(op.getAlias())) {
+    public ResponseCodeEnum validateEvmAddressAlias(final ByteString alias) {
+        if (HederaEvmContractAliases.isMirror(alias.toByteArray())) {
```

Comment:
Nullcheck for alias is missing

Reply:
yes, please do so.Since this is a public method somebody can call it with a null value

Reply:
I actually hadn't noticed that the method was public before --- it's private now